java-frontend/src/main/java/org/sonar/java/model/VisitorsBridge.java

```diff
@@ -115,8 +118,7 @@ public void visitFile(@Nullable Tree parsedTree) {
       scanner.scanFile(javaFileScannerContext);
     }
     if (semanticModel != null) {
-      // Close class loader after all the checks.
-      semanticModel.done();
+      ParametrizedTypeJavaType.typeSubstitutionSolver = null;
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Make the enclosing method "static" or remove this set. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2696)